src/object/cli_obj.c

```diff
@@ -2141,11 +2144,11 @@ obj_key_valid(daos_obj_id_t oid, daos_key_t *key, bool check_dkey)
 {
 	if (check_dkey) {
 		if (daos_is_dkey_uint64(oid) &&
-		    key->iov_len > sizeof(uint64_t))
+		    key->iov_len != sizeof(uint64_t))
```

Comment:
Minor, it looks we do print an error when this fails but it is not very specific as to which parameter was the problem.   Since we print an akey specific error, should we print a dkey one as well?